keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt

```diff
@@ -117,7 +117,7 @@ class MemoryCloudDriverCache(
           .map { it.availabilityZone }
           .toSet()
       }
-    }
+    }!!
 
   override fun subnetBy(subnetId: String): Subnet =
     subnets.getOrNotFound(subnetId, "Subnet with id $subnetId not found") {
```

Comment:
I believe `getOrNotFound` could be implemented using a `cache.get(key, function)` like your usages above.

Reply:
good point, thanks!